1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,244 @@
+package com.tridiumuniversity.myFile;
+
+import javax.baja.nre.annotations.NiagaraAction;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+
+
+@NiagaraType
+@NiagaraProperty(
+        name = "filename",
+        type = "String",
+        defaultValue = ""
+)
+
+@NiagaraProperty(
+        name = "path",
+        type = "String",
+        defaultValue = ""
+)
+
+@NiagaraProperty(
+        name = "size",
+        type = "int",
+        defaultValue = "0",
+        flags = Flags.SUMMARY
+)
+
+@NiagaraProperty(
+        name = "createdDate",
+        type = "BAbsTime",
+        defaultValue = "BAbsTime.NULL"
```

Comment:
They're technically the same thing, but `BAbsTime.DEFAULT` is probably a better choice for the default value. In general, I like to use the static `DEFAULT` field as the default value for all of my BSimple slots.